src/main/scala/mimir/statistics/FuncDep.scala

```diff
@@ -128,7 +128,7 @@ class FuncDep(config: Map[String,PrimitiveValue] = Map())
   // buildEntities calls all the functions required for ER creation, optionally each function could be called if only part of the computation is required
 
   def buildEntities(db: Database, query: Operator, tableName : String): Unit = {
-    initializeTables(db.bestGuessSchema(query), tableName)
+    initializeTables(db.schemaOf(query), tableName)
```

Comment:
db.typechecker.schemaOf (although I really suspect that it might be able to get away w/ just columns).